### PR TITLE
Improve re-login flow. Fixes: #2778

### PR DESCRIPTION
**Pull request by jainkuniya**
https://cl.ly/587235c377db

### CODE REVIEW

File: src/account/AccountItem.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* @flow */
```
Comment (borisyankov) [inline src/account/AccountItem.js]: > This resulted in too much automatically pushing and popping of screens, which doesn't look good.

Describe this in more detail. I think you might be referring to a behavior that is intended.

Comment (jainkuniya) [reply src/account/AccountItem.js]: Currently on clicking account item, we navigate to realm screen with auto fill realm url and tries to connect to it. So realm screen is pushed and within few seconds auth screen is pushed. 

On this PR, we are directly sending user to auth screen. Thus having less visual effects (sudden) which looks better. 

I guess we have discussed about this in India meet up :)

File: src/account/AccountPickScreen.js [hunk 2]
```diff
@@ -56,7 +56,7 @@ class AccountPickScreen extends PureComponent<Props> {
             text="Add new account"
             style={styles.button}
             onPress={() => {
-              dispatch(navigateToRealmScreen(''));
+              dispatch(navigateToRealmScreen());
```
Comment (borisyankov) [inline src/account/AccountPickScreen.js]: Good, we don't need that indeed, as we make sure to initialize the variable with '' if missing in `RealmScreen`.

File: src/account/AccountPickScreen.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ import type { Auth, Account, Dispatch, GlobalState } from '../types';
 import { getAuth, getAccounts } from '../selectors';
 import { Centerer, ZulipButton, Logo, Screen } from '../common';
 import AccountList from './AccountList';
-import { navigateToAddNewAccount, switchAccount, removeAccount } from '../actions';
+import { navigateToRealmScreen, switchAccount, removeAccount } from '../actions';
```
Comment (borisyankov) [inline src/account/AccountPickScreen.js]: Good thinking. I like the rename.

**Comment by borisyankov** [comment]
@gnprice I think the first two commits are a nice refactor and are independent of the 3d & 4th one and can be merged right now.
I am not completely convinced in the changes in the 3rd commit and will think about it again later.

File: src/account/AccountItem.js [hunk 4]
```diff
@@ -63,7 +64,9 @@ export default class AccountItem extends PureComponent<Props> {
             <RawLabel style={styles.text} text={email} numberOfLines={1} />
             <RawLabel style={styles.text} text={realm} numberOfLines={1} />
           </View>
-          {!showDoneIcon ? (
+          {showProgress ? (
```
Comment (borisyankov) [inline src/account/AccountItem.js]: I am usually fine with a single ternary operator, but two of them usually make the logic too messy.

**Comment by gnprice** [comment]
Merged those first two! ab9b52a2f.

Thanks @jainkuniya , and thanks @borisyankov for the review :smile:

File: src/account/AccountItem.js (hunk 4 above)
Comment (jainkuniya) [reply src/account/AccountItem.js]: Moved this logic to `AccountList`

**Comment by gnprice** [comment]
I've just filed #3685 for the underlying issue this is aimed at, which caused me to run across this PR again.

I don't think the code in this PR is the right direction -- it duplicates most of the logic that's in `tryRealm`, and that's a tricky stateful interaction that's well beyond the complexity of what I don't mind having duplicated.

Closing the PR. We'll use #3685 to continue to track the underlying issue.

**Comment by gnprice** [comment]
Also, for cross-reference's sake: the original issue description this referred to was #2778 . (It's in a commit message, and the PR title... but neither of those cause GitHub to provide a link from the issue back to the PR.)

**Pull request closed without merging**